### PR TITLE
MOD function related changes

### PR DESCRIPTION
**Pull request by naveenantonie**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 1]
```diff
@@ -756,6 +761,36 @@ && isIntervalHourAndSecond(call)) {
...
+    SqlCall modFunctionCall = MOD.createCall(SqlParserPos.ZERO, modifiedNodes);
+    MOD.unparse(writer, modFunctionCall, leftPrec, rightPrec);
+  }
+
+  private List<SqlNode> getModifiedModOperands(List<SqlNode> operandList) {
+    List<SqlNode> modifiedOperandList = new ArrayList<>();
+    for (SqlNode node : operandList) {
+      boolean isOperandNumericLiteral = node instanceof SqlNumericLiteral;
+      if (isOperandNumericLiteral) {
+        int precision = ((SqlNumericLiteral) node).getPrec();
```
Comment (InderjeetVishnoi) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:776]: This entire casting logic could be extracted into a seperate method, say `castToDecimalIfNeeded`

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 2]
```diff
@@ -2490,6 +2490,17 @@ private void checkHavingAliasSameAsColumn(boolean upperAlias) {
     sql(query).withHive().ok(expected);
   }
 
+  @Test void testModFunctionWithDecimalValues() {
```
Comment (InderjeetVishnoi) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:2493]: `testModFunctionWithNumeric/DecimalLiterals `would be better.

**Comment by InderjeetVishnoi** [comment]
LGTM

**Review by dm-shantanuc — APPROVED** [review]
LGTM

**Pull request merged**